File: 06/Lua/partOne.lua
local ta = {}
for line in io.lines("input.txt") do
    local numbers = {}
    for val in line:gmatch("(%d+)") do
        numbers[#numbers + 1] = tonumber(val)
    end
    ta[#ta + 1] = numbers
end
local ways = 1
-- distance + 1 > distance =  (totalTime - holdedTime) * holdedTime
-- distance + 1 > holdedTime * holdedTime - holdedTime ^ 2
-- holdedTime ^ 2 - totalTime * holdedTime + distance + 1 > 0
-- holdedTime > ( totalTime +- sqrt( totalTime ^ 2 - 4 * (distance + 1) ) ) / 2
for index = 1, #ta[1] do
    local time = ta[1][index]
    local distance = ta[2][index] + 1
    local t1 = math.floor((time + math.sqrt((time ^ 2) - (4 * distance))) / 2)
    local t2 = math.ceil((time - math.sqrt((time ^ 2) - (4 * distance))) / 2)
    ways = ways * (t1-t2 + 1)
end
print(ways)

File: 06/Python/partOne.py
import math


result = []
with open("input.txt", "r") as f:
    line = f.readlines()
    time = line[0].strip().split()[1:]
    distance = line[1].strip().split()[1:]
    s = 1
    for t,d in zip(time,distance):
        t = int(t)
        d = int(d)
        sums = 0
        # print(t/2)
        # print(math.floor(t/2)+1)
        for i in range(math.floor(t/2)+1):
            # print(i)
            dis = i * (t-i)
            if dis > d:
                print(i, dis)
                sums += 1 if t%2 == 0 and i == t//2 else 2
        # print(sums)
        s *= sums
    print(s)
